### PR TITLE
network: ndisc - Allow to use ICMP6 rate limit from received RA

### PR DESCRIPTION
**Pull request by ssahani**
Probably fixes https://github.com/systemd/systemd/issues/28434

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -2636,6 +2636,14 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>UseICMP6RateLimit=</varname></term>
+        <listitem>
+          <para>Takes a boolean. When true, the ICMP6 rate limit received in the Router Advertisement will be set to ICMP6
+          ratelimit based on the advertisement. Defaults to true.</para>
```
Comment (keszybz) [inline man/systemd.network.xml]: "rate limit", to match the previous, correct, spelling.

File: src/network/networkd-ndisc.c [hunk 2]
```diff
@@ -352,6 +354,21 @@ static int ndisc_router_process_default(Link *link, sd_ndisc_router *rt) {
...
 
+        r = sd_ndisc_router_get_icmp6_ratelimit(rt, &icmp6_ratelimit);
+        if (r < 0)
+                log_link_debug(link, "Failed to get default router preference from RA: %m");
+
+        if (icmp6_ratelimit > 0 && link->network->ipv6_accept_ra_use_icmp6_ratelimit) {
+                _cleanup_free_ char *p = NULL;
+
+                if (asprintf(&p, "%u", icmp6_ratelimit) < 0)
+                        return log_oom();
```
Comment (keszybz) [inline src/network/networkd-ndisc.c]:
```C
char buf[DECIMAL_STR_MAX(unsigned)];
xsprintf(buf, "%u", icmp6_ratelimit);
```

**Review by keszybz — COMMENTED** [review]
Looks reasonable.

File: src/network/networkd-ndisc.c (hunk 2 above)
Comment (ssahani) [reply src/network/networkd-ndisc.c]: thanks

**Comment by bluca** [comment]
> Can you add a test case for this?

test case is still missing

**Comment by ssahani** [comment]
> > Can you add a test case for this?
> 
> test case is still missing

This param is not provided by dnsmasq. We cant feed this to RA. What we can do is we support this in sd-radv and use it for testing

**Comment by ssahani** [comment]
> > > Can you add a test case for this?
> > 
> > 
> > test case is still missing
> 
> This param is not provided by dnsmasq. We cant feed this to RA. What we can do is we support this in sd-radv and use it for testing

https://github.com/systemd/systemd/pull/28888

**Comment by keszybz** [comment]
OK, this seems simple enough to merge even without a test. We should be able to add one later.

**Pull request merged**